src/main/java/bio/terra/service/dataset/flight/update/DatasetSchemaUpdateAddTablesBigQueryStep.java

```diff
@@ -0,0 +1,51 @@
+package bio.terra.service.dataset.flight.update;
+
+import bio.terra.common.Table;
+import bio.terra.model.DatasetSchemaUpdateModel;
+import bio.terra.model.TableModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetDao;
+import bio.terra.service.dataset.DatasetTable;
+import bio.terra.service.tabulardata.google.bigquery.BigQueryDatasetPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+public class DatasetSchemaUpdateAddTablesBigQueryStep implements Step {
+  private final BigQueryDatasetPdao bigQueryDatasetPdao;
+  private final DatasetDao datasetDao;
+  private final UUID datasetId;
+
+  private final DatasetSchemaUpdateModel updateModel;
+
+  public DatasetSchemaUpdateAddTablesBigQueryStep(BigQueryDatasetPdao bigQueryDatasetPdao, DatasetDao datasetDao, UUID datasetId, DatasetSchemaUpdateModel updateModel) {
+    this.bigQueryDatasetPdao = bigQueryDatasetPdao;
+    this.datasetDao = datasetDao;
+    this.datasetId = datasetId;
+    this.updateModel = updateModel;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    Dataset dataset = datasetDao.retrieve(datasetId);
+    List<String> newTableNames = DatasetSchemaUpdateUtils.getNewTableNames(updateModel);
+    List<DatasetTable> tables = dataset.getTables().stream().filter(dt -> newTableNames.contains(dt.getName())).collect(Collectors.toList());;
+    bigQueryDatasetPdao.createTables(dataset, tables);
+    return StepResult.getStepResultSuccess();
+  }
+
+  @Override
+  public StepResult undoStep(FlightContext context) throws InterruptedException {
+    Dataset dataset = datasetDao.retrieve(datasetId);
+    List<String> newTableNames = DatasetSchemaUpdateUtils.getNewTableNames(updateModel);
+    for (String tableName : newTableNames) {
+      bigQueryDatasetPdao.deleteDatasetTable(dataset, tableName);
```

Comment:
Just making sure: does this cause a failure if the table doesn't exist?  Could happen if one of the table creations fail

Reply:
Should the intended behavior here be to succeed the undo step even if one of the new tables doesn't exist and therefore can't be deleted?

Reply:
I think that makes sense, otherwise we wouldn't delete the rest of the tables right?